### PR TITLE
standardize log_dir for loggers

### LINKED ISSUE #7542: Cli with TestTubeLogger does not save the configuration file to the version folder
## 🐛 Bug

Cli with TestTubeLogger does not save the configuration file to the version folder

## Please reproduce using the BoringModel
```shell
python train.py
```

train.py: 
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import os

import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer
from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
from pytorch_lightning.utilities.cli import LightningCLI

class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)
    def __getitem__(self, index):
        return self.data[index]
    def __len__(self):
        return self.len

class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)
    def forward(self, x):
        return self.layer(x)
    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}
    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)
    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)
    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)
    def train_dataloader(self):
        return DataLoader(RandomDataset(32, 64), batch_size=2)
    def val_dataloader(self):
        return DataLoader(RandomDataset(32, 64), batch_size=2)
    def test_dataloader(self):
        return DataLoader(RandomDataset(32, 64), batch_size=2)

def run():
    early_stopping = EarlyStopping(monitor="valid_loss")
    checkpoint_callback = ModelCheckpoint(dirpath="logs", monitor="valid_loss")

    cli = LightningCLI(
        BoringModel,
        seed_everything_default=123,
        trainer_defaults={
            "max_epochs": 2,
            "callbacks": [
                checkpoint_callback,
                early_stopping,
            ],
            "logger": {
                "class_path": "pytorch_lightning.loggers.TestTubeLogger",
                "init_args": {
                    "save_dir": "logs",
                    "create_git_tag": True,
                },
            },
        },
    )
    cli.trainer.test(cli.model)

if __name__ == '__main__':
    run()
```

### Expected behavior

config files are saved to `logs/version_d`.

### Environment

```
* CUDA:
	- GPU:
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
		- TITAN RTX
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.2
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu102
	- pytorch-lightning: 1.3.1
	- tqdm:              4.50.2
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.5
	- version:           #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021
```

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7542

`logger.log_dir` and `trainer.log_dir` was not consistently pointing to the experiment folder. It means user code cannot be logger agnostic and also the CLI has no fixed place where config files land. This PR introduces `logger.experiment_dir` for all loggers and deprecates the old `logger.log_dir` property.

TODO: 
- [x] update tests
- [ ] deprecation message

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/properties.py [hunk 1]
```diff
@@ -139,6 +144,12 @@ def log_dir(self) -> Optional[str]:
         dirpath = self.accelerator.broadcast(dirpath)
         return dirpath
 
+    @property
+    def experiment_dir(self):
+        dirpath = self.logger.experiment_dir if self.logger is not None else self.default_root_dir
+        dirpath = self.accelerator.broadcast(dirpath)
```
Comment (ananthsub) [inline pytorch_lightning/trainer/properties.py:150]: this broadcast should be done once so we're not calling this on every property access

**Comment by tshu-w** [comment]
ping for this 👀

**Comment by tshu-w** [comment]
ping for this 👀 I still cannot use other loggers with CLI.
